ui/dashboardApp/index.ts

```diff
@@ -72,15 +81,38 @@ async function main() {
     if (spinner) {
       spinner.remove()
     }
-    if (!routing.isLocationMatchPrefix(auth.signInRoute)) {
+    if (!routing.isSignInPage()) {
       if (!auth.getAuthTokenAsBearer()) {
-        singleSpa.navigateToUrl('#' + auth.signInRoute)
-        return
+        singleSpa.navigateToUrl('#' + routing.signInRoute)
       }
     }
   })
 
   singleSpa.start()
 }
 
-main()
+/////////////////////////////////////
+
+function start() {
+  // the portal page is only used to receive options
+  if (routing.isPortalPage()) {
+    function handleConfigEvent(event) {
+      const { token, lang, hideNav, app } = event.data
+      auth.setAuthToken(token)
+      saveAppOptions({ hideNav, lang })
+
+      // redirect
+      const { origin, pathname } = window.location
+      const appUrl = `${origin}${pathname}#/${app || 'statement'}`
+      window.location.href = appUrl
```

Comment:
how about `window.location.hash = `#${redirectPath}` directly

Reply:
Don't know this way before, not sure whether it works, let me try it.

Reply:
Awesome! works perfectly.